packages/vite-plugin-cloudflare/src/workers-configs.ts

```diff
@@ -326,30 +331,40 @@ export function getWorkerConfig(
 		throw new Error(missingFieldErrorMessage(`'main'`, configPath, env));
 	}
 
-	const mainStat = fs.statSync(config.main, { throwIfNoEntry: false });
-	if (!mainStat) {
-		throw new Error(
-			`The provided Wrangler config main field (${config.main}) doesn't point to an existing file`
-		);
-	}
-	if (mainStat.isDirectory()) {
-		throw new Error(
-			`The provided Wrangler config main field (${config.main}) points to a directory, it needs to point to a file instead`
-		);
-	}
-
 	return {
 		type: "worker",
 		raw,
 		config: {
 			...config,
 			...requiredFields,
-			main: config.main,
+			main: maybeResolveMain(config.main, configPath),
 		},
 		nonApplicable,
 	};
 }
 
+/**
+ * If `main` ends with a valid file extension it is resolved to an absolute path.
+ * Else `main` is returned as is so that it can be resolved by Vite.
+ * This enables resolving entry modules relative to the Worker config while also supporting virtual modules and package exports.
+ */
+function maybeResolveMain(main: string, configPath: string): string {
+	if (!ENTRY_MODULE_EXTENSIONS.some((extension) => main.endsWith(extension))) {
+		return main;
+	}
```

Comment:
You can technically include an extension in a package entrypoint, though that seems pretty uncommon. Would it be safer to default to treating it as a path, and only consider it a module if it does not exist on the file system?

Reply:
The issue then is that it would be a change to the existing behaviour where we error if the file does not exist. There's also a slight risk that the file at the absolute path doesn't exist so we pass it through but then it resolves to a different (and unintended) file that does exist. Part of the problem is that we resolve the `main` field relative to the Worker config file and we treat paths as relative, even if they don't have leading `./`.

Not allowing `package.json` exports that include a file extension seems like an OK tradeoff but open to other ideas here.